hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/BlockOutputStreamEntryPool.java

```diff
@@ -368,7 +371,16 @@ void hsyncKey(long offset) throws IOException {
       if (keyArgs.getIsMultipartKey()) {
         throw new IOException("Hsync is unsupported for multipart keys.");
       } else {
-        omClient.hsyncKey(keyArgs, openID);
+        if (keyArgs.getLocationInfoList().size() == 0) {
+          omClient.hsyncKey(keyArgs, openID);
+        } else {
+          ContainerBlockID lastBLockId = keyArgs.getLocationInfoList().get(keyArgs.getLocationInfoList().size() - 1)
+              .getBlockID().getContainerBlockID();
+          if (!lastUpdatedBlockId.equals(lastBLockId)) {
+            omClient.hsyncKey(keyArgs, openID);
```

Comment:
One question -- when a new block is allocated, the key is updated with the new block, right? So why is this step still necessary?

Reply:
The key is updated with new block, new block with the length at the time when hsync is called. The later hsync on this block will not update the new block length of key in OM. So the last block length of key in OM is not update to date.

Reply:
I see.